Improved space names

prabau opened the pull request:
Improved names and streamlined aliases for some spaces, as discussed in latest zoom meeting.

My guiding principle when removing some aliases was that, due to fuzzy matching, the space would still be accessible by starting to type the previous alias.

prabau commented:
Please compare the aliases for S28 and S102 ("countable power ...", "countable product ...") and see if you want some change there.

pzjp commented on spaces/S000025/README.md:
The $\mathbb{R}$ here seems redundant. Adding it to the alias "Euclidean topology" is justified, though.

pzjp commented:
I agree with adding the underlying sets to some names (or changing long "real numbers" to "R" where possible) but I am not sure whether we should append symbols to otherwise complete names, like "Rational numbers $\mathbb Q$". Is there any ambiguity behind "rational numbers"? I think that the symbol may be good as a standalone alias (they used to be aliases, at least sometimes). Or they cause some search problems?

pzjp commented:
So we start with (almost) justified case like **Cantor space $\\{0,1\\}^\omega$** and extend this convention? How many "good examples" there are?
Sorry, if I just do not get something language-wise.

prabau commented:
@pzjp @StevenClontz (some questions for you further down)

My general thinking is that some spaces have a canonical/well-known symbol, and it seems useful to have this as part of the pi-base name.

For an example, S34 used to have "Successor to the second limit ordinal" as the main name.  At one point we changed it to 
"Ordinal space $\omega+\omega+1$".

That makes the space more easily recognizable.  It's particularly useful when we refer to the space with the syntax `{S34}` in the middle of the text for a proof of a trait or a description of some other space, etc.  Having the symbol in there makes a long paragraph of text more digestible somehow.

----
As for the aliases, they are less important.  Sometimes they refer to a non-obvious homeomorphic description of the space (e.g. S28: Baire space = Irrational numbers, S30: Hilbert space $\ell^2$ = countable product of reals = etc).  So useful in that case.

Sometimes aliases basically describe the same thing as the main name, just with a different wording.  I think some of that could be streamlined.  Sometimes an alias is there just because it was the name used in Steen & Seebach, but even in that case, thanks for fuzzy matching in the searches, we may not need to keep that particular alias in some cases.

Example S25 before this PR:
Main name: "Euclidean Real Numbers"
Aliases: Real line, Euclidean topology, Euclidean space

"Euclidean topology" was there because it was the name from S&S.  Not very good, because it's conflating the topology and the space, and also not unique enough (but that does not matter for an alias).  With fuzzy matching, it's enough to type "Euclidean" and this spaces shows up at the top of the list.  **Therefore can we remove this alias as well?**

You were suggesting to use just $\mathbb R$ for the main name.  That's a possibility.  But $\mathbb R$ is just the set of reals and there are many topologies on it.  That's why having the word "Euclidean" in there seems worth it (and it also allows to remove the alias "Euclidean topology").  But **maybe it would be ok without Euclidean too**, as "Real line $\mathbb R$ " for the name, since the Euclidean topology is the common understanding is nothing else is specified?
Thoughts?

Example S16 (not part of this PR, but worth discussing)
Main name: Cofinite topology on $\mathbb R$
Aliases:
- Finite complement topology on the real numbers
- Finite Complement Topology on an Uncountable Space
- Cofinite topology on the real numbers

It's good to have "cofinite" and to also have "finite complement" for searching purposes.  If we had "Finite complement topology on $\mathbb R$" as alias and removed the other aliases, the searches would still work fine and the list would be more manageable.  Thoughts?

StevenClontz commented:
My biggest concern with changing aliases has been that it could end up breaking search URLs used across the site or on external forums. But, thinking a beat just now, adding aliases (or any new entry with a name/alias) can do this as well. One day we should have permanent URLs for things like "show me all P1+P2+~P3" spaces, regardless of whether those names/aliases change. But we can't solve that here, and I think I'm in favor of a general cleanup of names/aliases without regards for accidentally breaking old searches (but keeping any reasonable keyword somewhere in a name/alias).
